### PR TITLE
ADD KMS CMEK encryption option for GCS bucket creation

### PR DESCRIPTION
**Pull request by hayorov**
- Add an option to create GCP buckets with KMS /CMEK encryption option
- Pre-created KMS key is required
- Default is the same (no encryption)
- Minor go version and fmt updates

### CODE REVIEW

File: deploy/overlays/stable/kustomization.yaml [hunk 1]
```diff
@@ -4,4 +4,4 @@ bases:
 - ../../base
 images:
 - name: docker.io/ofekmeister/csi-gcs
-  newTag: v0.5.0
```
Comment (maennchen) [inline deploy/overlays/stable/kustomization.yaml:7]: I would prefer to do the release in a separate commit.

**Review by maennchen — CHANGES_REQUESTED** [review]

**Comment by maennchen** [comment]
@hayorov Thanks a lot for your contribution.

I'd like to await a review from @ofek as well, in my own opinion it looks good.

**Review by ofek — CHANGES_REQUESTED** [review]
@hayorov Great job, thanks!

Let's remove the version bump as @maennchen suggested, then I'll merge.

**Comment by hayorov** [comment]
Many thanks, @maennchen, @ofek.

Updated according to comments.

**Review by ofek — APPROVED** [review]
@maennchen Do you mind releasing?

**Comment by maennchen** [comment]
@ofek It's released. There's a problem though with the release of the docs: https://github.com/ofek/csi-gcs/issues/62

**Pull request merged**